tests/test-misc.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+class VIP_Go_Misc_Test extends WP_UnitTestCase {
```

Comment:
I wonder if we should just make this a class + file specific to the `check_for_404_and_remove_cache_headers` function. That way, we can utilize `setUp` and/or `tearDown` functions instead of relying on cleanup within the test itself.